jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -135,21 +151,36 @@ public void doWrite(Server server, Query query, ImmutableList<Result> results) t
 		influxDB.createDatabase(database);
 
 		BatchPoints batchPoints = BatchPoints.database(database).retentionPolicy(retentionPolicy)
-				.tag(TAG_HOSTNAME, server.getHost()).consistency(writeConsistency).build();
+				.tag(TAG_HOSTNAME, server.getSource()).consistency(writeConsistency).build();
+
 		for (Result result : results) {
-			Map<String, String> resultTagsToApply = buildResultTagMap(result);
-			Point point = Point.measurement(result.getKeyAlias()).time(result.getEpoch(), MILLISECONDS)
-					.tag(resultTagsToApply).fields(result.getValues()).build();
-			batchPoints.point(point);
+
+			HashMap<String, Object> filteredValues = new HashMap(Maps.filterValues(result.getValues(), isNotNaN));
+			filteredValues.put("_jmx_port", Integer.parseInt(server.getPort()));
+
+			// send the point if filteredValues isn't empty
+			if (filteredValues.size() > 1) {
```

Comment:
I read that too fast ... I'd prefer moving the part about `_jmx_port` inside the `if`. That way we can use an `isEmpty()` which is a bit more clear IMHO. Does it make sense to you ?